src/repositories/UniswapV3PoolRepository.ts

```diff
@@ -0,0 +1,46 @@
+import {injectable} from 'inversify';
+import {getModelForClass} from '@typegoose/typegoose';
+
+import {UniswapV3Pool} from '../models/Pools.js';
+import {ChainsIds} from '../types/ChainsIds.js';
+
+@injectable()
+export class UniswapV3PoolRepository {
+  async save(
+    props: Omit<
+      UniswapV3Pool,
+      'createdDate' | 'liquidityActive' | 'liquidityLockedToken0' | 'liquidityLockedToken1' | 'liquidityUpdatedAt'
+    >,
+  ): Promise<UniswapV3Pool> {
+    const UniswapV3PoolModel = getModelForClass(UniswapV3Pool);
+
+    const uniswapV3PoolData = new UniswapV3PoolModel({
+      ...props,
+      createdDate: new Date(Date.now()),
+    });
+
+    return uniswapV3PoolData.save();
+  }
+
+  async saveLiquidity(
+    filter: {chainId: ChainsIds; token0: string; token1: string; fee: number; address: string},
+    liquidity: {liquidityActive: number; liquidityLockedToken0: number; liquidityLockedToken1: number},
+  ): Promise<UniswapV3Pool | null> {
+    return getModelForClass(UniswapV3Pool)
+      .findOneAndUpdate(
+        filter,
+        {...liquidity, liquidityUpdatedAt: new Date(Date.now())},
+        {
+          new: true,
+        },
+      )
+      .exec();
+  }
+
+  async find(props: {protocol: string; fromChain: string[]; token0: string; token1: string}): Promise<UniswapV3Pool[]> {
+    const {protocol, fromChain, token0, token1} = props;
+    const filter = {protocol, chainId: {$in: fromChain}, token0, token1};
```

Comment:
before we use tokens in filter, please sort them, so token0 < token1

Reply:
DB:
1. pool: a, token0: 0x123, token1: 0x234
2. pool: b, token0: 0x123, token1: 0x234

`find({token1: 0x123, token0: 0x234})`
`find({token0: 0x123, token1: 0x234})`

 for both `find` result should be 2 records.

Reply:
I added a sort in the find, It would be sort token0 and token1

Reply:
Sorry, maybe I got wrong.
If I do a find
```find({token1: 0x123, token0: 0x234})```
should we return 2 rows, one for pool A and another for pool B? Because token0 could be placed in token1 in DB?

Should we consider if the pool address is duplicated but the token0 and token1 are swapped?

Reply:
in example you have 2 pools in DB (different pools for the same tokens)

And 2 examples of `find` (they are different). Goal is both of them should return both pools (a and b).

Reply:
Okay, so I think we are good here, I changed to use $in, so it will return both now.
I am considering that we won't have token0 = token1

Reply:
pool: a, token0: 0x123, token1: 0x234
pool: b, token0: 0x123, token1: 0x234
pool: c, token0: 0x123, token1: 0x233

find({token1: 0x123, token0: 0x234}) => result (a, b)
find({token0: 0x123, token1: 0x234}) => result (a, b)

please create unit test.

Reply:
Done